pages/DynamicPOC/index.tsx

```diff
@@ -76,71 +77,85 @@ const DynamicPOC: React.FC = () => {
   ) {
     const answer = {
       questionId: event.target.name,
-      answerId: isUserInput
-        ? _getInputAnswerId(event.target.name)
-        : event.target.value,
+      answerId: isUserInput ? '0' : event.target.value,
       userAnswer: isUserInput ? event.target.value : undefined,
     }
 
     setCurrentAnswer(answer)
   }
 
-  function _getInputAnswerId(questionId: string): string {
-    const question: Question = questions[
-      questionId as QuestionsKeys
-    ] as Question
-    return question.answers[0].toString() // TODO: Probably error check this or something
-  }
-
   /**
    * handles getting the next question based on current question's answer
    */
   function _handleNext() {
-    if (formValues.formAnswers.hasOwnProperty(currentAnswer.questionId)) {
+    if (
+      !currentAnswer ||
+      currentAnswer.questionId !== currentQuestion.id.toString()
+    ) {
+      return
```

Comment:
What is this returning? I feel like I'm misreading this and it seems like it isn't returning anything.

Reply:
Correct - `_handleNext()` has a return type of `void`. This `if` just ensures that if the user has not selected an answer we don't let them progress, by returning out of handleNext entirely.